pkg/services/rendering/plugin_mode.go

```diff
@@ -45,11 +46,12 @@ func (rs *RenderingService) renderViaPluginV1(ctx context.Context, renderKey str
 	rs.log.Debug("calling renderer plugin", "req", req)
 
 	rsp, err := rs.pluginInfo.GrpcPluginV1.Render(ctx, req)
-	if ctx.Err() == context.DeadlineExceeded {
-		rs.log.Info("Rendering timed out")
-		return nil, ErrTimeout
-	}
 	if err != nil {
+		if errors.Is(ctx.Err(), context.DeadlineExceeded) {
+			rs.log.Info("Rendering timed out")
+			return nil, ErrTimeout
+		}
```

Comment:
Should this have been lifted in the `if err != nil`? Is `err` guaranteed to not be `nil` if `ctx.Err()` is set?

Reply:
Good point @sakjur. Looking again.

Reply:
@sakjur Good catch, it was a mistake!